src/theory/quantifiers/relevant_domain.cpp

```diff
@@ -169,7 +169,9 @@ void RelevantDomain::computeRelevantDomain( Node q, Node n, bool hasPol, bool po
         computeRelevantDomainOpCh( rf, n[i] );
       }
     }
-    if( n[i].getKind()!=FORALL ){
+    // do not recurse under nested quantifiers/lambdas
+    if (n[i].getKind() != FORALL && n[i].getKind() != LAMBDA)
```

Comment:
Why don't we add a general utility function `isBinder` that is called in different places of the code whenever you should have such a test?

Reply:
Oh, strange, we aren't have it: Node::isClosure.  I'll change to use this.

Reply:
Funny this is not used anywhere. BTW shouldn't `isClosure` also include choice and lets?

Reply:
There is no such thing as a let. I've already updated it to include choice.